app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/close/SetClosingDate.tsx

```diff
@@ -44,16 +47,44 @@ export const SetClosingDate = ({
   }, [closedMessage]);
 
   const [status, setStatus] = useState(closingDate ? "closed" : "open");
+  const [showDateTimeDialog, setShowDateTimeDialog] = useState(false);
 
   const handleToggle = (value: boolean) => {
     setStatus(value == true ? "closed" : "open");
   };
 
+  const saveFutureDate = useCallback(
+    async (futureDate?: number) => {
+      if (!futureDate) {
+        return;
+      }
+
+      const closingDate = new Date(futureDate).toISOString();
+
+      const result = await closeForm({
+        id: formId,
+        closingDate,
+        closedDetails: closedMessage,
+      });
+
+      if (!result || result.error) {
+        toast.error(t("closingDate.savedErrorMessage"));
+        return;
+      }
+
+      // Update the local template store
+      setClosingDate(closingDate);
+
+      toast.success(t("closingDate.savedSuccessMessage"));
+    },
+    [formId, setClosingDate, t, closedMessage]
+  );
+
   const saveFormStatus = useCallback(async () => {
-    let closeDate = "open";
+    let closeDate = "open"; // this wil reset the closing date to null;
```

Comment:
We can pass just null here and update the server action but we need to check of the closing date is alresdy set in the futrure before we null out the date.